src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -293,6 +294,48 @@ public Response downloadZip(JsonObject content,
       }
    }
 
+   @POST
+   @javax.ws.rs.Path("/commands/{commandName}/catapult")
+   @Consumes(MediaType.APPLICATION_JSON)
+   public Response uploadZip(JsonObject content,
+                               @PathParam("commandName") @DefaultValue(DEFAULT_COMMAND_NAME) String commandName,
+                               @Context HttpHeaders headers)
+         throws Exception
+   {
+      Response response = downloadZip(content, commandName, headers);
+      if (response.getStatus() != 200) {
+         return response;
+      }
+
+      String fileName = findArtifactId(content);
+      byte[] zipContents = (byte[]) response.getEntity();
+
+      UriBuilder uri = UriBuilder.fromPath("/api/catapult/upload");
+      uri.host(System.getenv("CATAPULT_SERVICE_HOST"));
```

Comment:
don't know if it always there if you deploy first the generator backend end then catapult. Don't want this service to explode if catapult is not there. Not sure what the best default action would be